src/FastRouteRouter.php

```diff
@@ -80,15 +109,31 @@ class FastRouteRouter implements RouterInterface
      *     implementation will be used.
      * @param null|callable $dispatcherFactory Callable that will return a
      *     FastRoute dispatcher.
+     * @param array $config Array of custom configuration options.
      */
-    public function __construct(RouteCollector $router = null, callable $dispatcherFactory = null)
-    {
+    public function __construct(
+        RouteCollector $router = null,
+        callable $dispatcherFactory = null,
+        array $config = null
+    ) {
         if (null === $router) {
             $router = $this->createRouter();
         }
 
         $this->router = $router;
         $this->dispatcherCallback = $dispatcherFactory;
+
+        if (isset($config['cache_enabled'])) {
+            $this->cacheEnabled = (bool) $config['cache_enabled'];
```

Comment:
Please use class constants, not just magic strings. We had enough typos due to these in the past